earthaccess/__init__.py

```diff
@@ -36,7 +37,30 @@
     "Store",
 ]
 
-__auth__ = Auth()
-__store__: Any = None
-
 __version__ = version("earthaccess")
+
+_auth = Auth()
+_store = None
+_lock = threading.Lock()
+
+
+def __getattr__(name):  # type: ignore
+    global _auth, _store
+
+    if name == "__auth__" or name == "__store__":
+        with _lock:
+            if not _auth.authenticated:
+                for strategy in ["environment", "netrc"]:
+                    try:
+                        _auth.login(strategy=strategy)
+                    except Exception:
+                        continue
+                    else:
+                        if not _auth.authenticated:
+                            continue
```

Comment:
The "environment" strategy will currently log at debug level in this case, which would be triggered if envvars are not set. The "netrc" strategy, however, [will not](https://github.com/jrbourbeau/earthaccess/blob/641eafcf1ae7807cd15323502eb7dab47133ee11/earthaccess/auth.py#L248-L250). Should we have more debug logging on this functionality, since it's happening implicitly?

Maybe we should debug log here "Automatic login attempted with {strategy=} but did not succeed"?  It couldn't hurt to debug log in the `except` case, too. And maybe also in the successfully-authenticated with `{strategy}` case?

Reply:
Went ahead and added more debug logging here. I think there's enough information now to help figure out where things go wrong (if they do go wrong) but let me know if there are some cases I missed

Reply:
I think it looks great, thanks James :)